app/assets/javascripts/forever_style_guide/hero-blur.js

```diff
@@ -0,0 +1,422 @@
+          }
+          var attachment = $source.css('backgroundAttachment');
+          var position = (attachment == 'fixed') ? '' : '-' + (($glue.offset().left) - ($source.offset().left) - (options.offset.x)) + 'px -' + (($glue.offset().top) - ($source.offset().top) - (options.offset.y)) + 'px';
+          $glue.css({
+            'background-image': 'url("' + blurredData + '")',
+            'background-repeat': $source.css('backgroundRepeat'),
+            'background-position': position,
+            'background-attachment': attachment
+          });
+          if(options.optClass != false) {
+            $glue.addClass(options.optClass);
+          }
+          if(options.draggable) {
+            $glue.css({
+              'background-attachment': 'fixed',
+              'background-position': '0 0'
+            });
+            $glue.draggable();
+          }
+
+          $glue.attr({
+            "data-hero-width" : canvas.width,
+            "data-hero-height" : canvas.height,
+            "data-hero-overlay": options.overlay
+          });
+          $glue.trigger( "loaded", canvas);
+        };
+        Storage.prototype.cacheChecksum = function (opts) {
+          var newData = '';
+          for(var key in opts) {
+            var obj = opts[key];
+            if(obj.toString() == '[object Object]') {
+              newData += ((obj.x).toString() + (obj.y).toString() + ",").replace(/[^a-zA-Z0-9]/g, "");
+            } else {
+              newData += (obj + ",").replace(/[^a-zA-Z0-9]/g, "");
+            }
+          }
+          var originalData = this.getItem(options.cacheKeyPrefix + selector + '-' + formattedSource + '-options-cache');
+          if(originalData != newData) {
+            this.removeItem(options.cacheKeyPrefix + selector + '-' + formattedSource + '-options-cache');
+            this.setItem(options.cacheKeyPrefix + selector + '-' + formattedSource + '-options-cache', newData);
+            if(options.debug) {
+              console.log('Settings Changed, Cache Emptied');
+            }
+          }
+        };
+        var cachedData = null;
+        if(options.cache) {
+          localStorage.cacheChecksum(options);
+          cachedData = localStorage.getItem(options.cacheKeyPrefix + selector + '-' + formattedSource + '-data-image');
+        }
+        if(cachedData != null) {
+          if(options.debug) {
+            console.log('Cache Used');
+          }
+          isCached = true;
+          tempImg.src = (cachedData);
+          canvas.width = tempImg.width;
+          canvas.height = tempImg.height;
+        } else {
+          if(options.debug) {
+            console.log('Source Used');
+          }
+          tempImg.src = formattedSource;
+        }
+      });
+    };
+    return $; // return jQuery
+  });
+})(
+  typeof define === 'function' && define.amd
+    ? define
+    : function (r, factory) { factory(jQuery); }
+);
+
+// percentage of 'x' position of center of focus for bg scaling
```

Comment:
So this is totally on me, but because we are including this into the style guide now we probably don't want to expose these as the global variables they are here.  It would be preferable to wrap them and the functions below into some sort of module closure.